Sources/SwiftGenKit/Parsers/Strings/StringsParser.swift

```diff
@@ -38,36 +44,76 @@ public enum Strings {
 
   public final class Parser: SwiftGenKit.Parser {
     private let options: ParserOptionValues
+    private var parsers = [String: StringsFileTypeParser.Type]()
     var tables = [String: [Entry]]()
     public var warningHandler: Parser.MessageHandler?
 
+    private static let subParsers: [StringsFileTypeParser.Type] = [
+      StringsFileParser.self,
+      StringsDictFileParser.self
+    ]
+
     public init(options: [String: Any] = [:], warningHandler: Parser.MessageHandler? = nil) throws {
       self.options = try ParserOptionValues(options: options, available: Parser.allOptions)
       self.warningHandler = warningHandler
+
+      for parser in Parser.subParsers {
+        register(parser: parser)
+      }
     }
 
-    public static let defaultFilter = "[^/]\\.strings$"
     public static let allOptions: ParserOptionList = [Option.separator]
+    public static var defaultFilter: String {
+      let extensions = Parser.subParsers.flatMap { $0.extensions }.sorted().joined(separator: "|")
+      return "[^/]\\.(?i:\(extensions))$"
+    }
 
     // Localizable.strings files are generally UTF16, not UTF8!
     public func parse(path: Path, relativeTo parent: Path) throws {
+      guard let parserType = parsers[path.extension?.lowercased() ?? ""] else {
+        throw ParserError.unsupportedFileType(path: path, supported: parsers.keys.sorted())
+      }
+
       let name = path.lastComponentWithoutExtension
+      let tableExists = tables[name] != nil
+      let parser = parserType.init(options: options)
 
-      guard tables[name] == nil else {
-        throw ParserError.duplicateTable(name: name)
-      }
-      guard let data = try? path.read() else {
-        throw ParserError.failureOnLoading(path: path.string)
-      }
+      let entries: [Entry]
+      if tableExists {
+        guard parser.isAllowedToWriteToExistingTable else {
+          throw ParserError.duplicateTable(name: name)
+        }
 
-      let plist = try PropertyListSerialization.propertyList(from: data, format: nil)
-      guard let dict = plist as? [String: String] else {
-        throw ParserError.invalidFormat
+        let existingEntries = tables[name] ?? []
+        let newEntries = try parser.parseFile(at: path)
+        let mergedEntries = Array(Dictionary([existingEntries, newEntries].joined().map { ($0.key, $0) },
+                                             uniquingKeysWith: { $1 }).values)
```

Comment:
I did a few tests with Attabench, and the Dictionary initializer is way more performant than the variant using `reduce`. Probably because with `reduce`, we are calling `contains` quite a lot, which is an _O(n)_ call. I split the call up into more variables, it should be more readable now